### PR TITLE
[CI] Adds support for selecting experiments for workflows on runner determinator

### PR DESCRIPTION
**Pull request by jeanschmidt**
adds a `default` tag to experiment configurations, allowing to remove some experiments by default on the random draw:

```
        experiments:
            lf:
                rollout_perc: 25
            otherExp:
                rollout_perc: 25
                default: false
        ---
```

and includes the configuration to filter what experiments are of interest for a particular workflow (comma separated):

```
  get-test-label-type:
    name: get-test-label-type
    uses: ./.github/workflows/_runner-determinator.yml
    with:
      ...
      check_experiments: "awsa100"
```

The end goal, is to enable us to run multiple experiments, that are independent from one another. For example, while we still runs the LF infra experiment, we want to migrate other runners leveraging the current solution. A immediate UC is for the A100 instances, where we want to migrate to AWS. 

Those new instances will during the migration period be labeled both `awsa100.linux.gcp.a100` and `linux.aws.a100`. Once the experiment ends, we will remove the first confusing one.

```
jobs:
  get-build-label-type:
    name: get-build-label-type
    uses: ./.github/workflows/_runner-determinator.yml
    with:
      ...

  get-test-label-type:
    name: get-test-label-type
    uses: ./.github/workflows/_runner-determinator.yml
    with:
      ...
      check_experiments: "awsa100"
      
  linux-focal-cuda12_1-py3_10-gcc9-inductor-build:
    name: cuda12.1-py3.10-gcc9-sm80
    uses: ./.github/workflows/_linux-build.yml
    needs:
      - get-build-label-type
      - get-test-label-type
    with:
      runner_prefix: "${{ needs.get-build-label-type.outputs.label-type }}"
      ...
      test-matrix: |
        { include: [
          { config: "inductor_huggingface_perf_compare", shard: 1, num_shards: 1, runner: "${{ needs.get-test-label-type.outputs.label-type }}linux.gcp.a100" },
          ...
        ]}
      ...
```

```
experiments:
    lf:
        rollout_perc: 50
    awsa100:
        rollout_perc: 50
         default: false
```

### CODE REVIEW

File: .github/scripts/runner_determinator.py [hunk 1]
```diff
@@ -174,6 +182,13 @@ def parse_args() -> Any:
         required=True,
         help="Current GitHub ref type, branch or tag",
     )
+    parser.add_argument(
+        "--check-experiments",
+        type=_str_comma_separated_to_set,
```
Comment (ZainRizvi) [inline .github/scripts/runner_determinator.py:187]: TIL: you can pass in a function here!

File: .github/scripts/runner_determinator.py [hunk 2]
```diff
@@ -377,8 +393,18 @@ def get_runner_prefix(
             )
             enabled = True
         elif experiment_settings.rollout_perc:
+            do_check = True
+            if check_experiments:
+                if experiment_name not in check_experiments:
+                    exp_list = ", ".join(check_experiments)
+                    log.info(f"Skipping experiment '{experiment_name}', as it is not in the check_experiments list: {exp_list}")
+                    do_check = False
```
Comment (ZainRizvi) [inline .github/scripts/runner_determinator.py:401]: When an experiment is not a default experiment and not a check-experiment, even if the user is explicitly opted in the experiment should be disabled for them

File: .github/scripts/runner_determinator.py [hunk 3]
```diff
@@ -377,8 +393,18 @@ def get_runner_prefix(
...
+            elif not experiment_settings.default:
```
Comment (ZainRizvi) [inline .github/scripts/runner_determinator.py:402]: if check experiments is enabled, then we should ignore the `default` setting and only use the `check` experiments settings.

Else you'll pull in the `lf` prefix as well

File: .github/scripts/runner_determinator.py [hunk 4]
```diff
@@ -174,6 +182,13 @@ def parse_args() -> Any:
         required=True,
         help="Current GitHub ref type, branch or tag",
     )
+    parser.add_argument(
+        "--check-experiments",
```
Comment (ZainRizvi) [inline .github/scripts/runner_determinator.py:186]: naming suggestion:

```suggestion
        "--eligible-experiments",
```

**Review by ZainRizvi — DISMISSED** [review]

File: .github/scripts/runner_determinator.py (hunk 2 above)
Comment (jeanschmidt) [reply .github/scripts/runner_determinator.py:401]: ohh good point...

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**